src/Weikio.ApiFramework.Core/Cache/DefaultEndpointCache.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Weikio.ApiFramework.Abstractions;
+using Weikio.ApiFramework.AspNetCore;
+
+namespace Weikio.ApiFramework.Core.Cache
+{
+    internal class DefaultEndpointCache : IEndpointCache
+    {
+        private readonly IApiCache _apiCache;
+        private readonly IHttpContextAccessor _contextAccessor;
+
+        public DefaultEndpointCache(IApiCache apiCache, IHttpContextAccessor contextAccessor)
+        {
+            _apiCache = apiCache;
+            _contextAccessor = contextAccessor;
+        }
+
+        public string GetOrCreateString(string key, Func<string> getString)
+        {
+            var endpoint = _contextAccessor.HttpContext.GetEndpointMetadata();
```

Comment:
Create a new private method `GetEndpoint` and throw `InvalidOperationException` if `_contextAccessor.HttpContext` is null. Use the new method in all other publi methods in this class.

Reply:
Done.